packages/contracts/plugins/hijack-compiler.ts

```diff
@@ -0,0 +1,34 @@
+import * as path from 'path'
```

Comment:
1. This is dope!!!!!! I'm excited to see if we can use buidler for our fullnode tests!
2. Should this live in this package? Would we ever use this plugin in the `contracts` package? My guess would be that if we do any transpilation in the contract unit tests, that we would only be transpiling directly & not hook into the low level of buidler. Does that vibe with your intuitions?

Reply:
Talked about it in a different forum -- Kelvin suggested / I think the move is to keep the plugin here for now, and then in a new PR move it into it's own package similar to how we have the truffle plugins in a different package -- https://github.com/ethereum-optimism/optimism-monorepo/tree/9b94f0d317b719d2f6b7b48a2eae0bc9c69c5307/packages/ovm-truffle-provider-wrapper

Reply:
Hey 👋 I'm one of the authors of Buidler, and just happened to found this browsing github. I'm really curious about this override. Do you do this to run `solc` nightlies, or do you use your own version/fork of `solc`? Do you have any idea about what would make this easier to achieve? Thanks!

Reply:
Hey @alcuadrado! We currently make use of a custom solidity compiler (translating certain L1-related op codes into operations that can be executed on L2). Since buidler only allows version strings by default, we had to make this slight modification to allow paths to custom solc wrappers. I think it'd be great to allow for path/custom config variables within the solc config.

Reply:
If this isn't something you can/want to support natively, an easier change could be to split the steps where the compiler is initialized and the files are compiled into two internal tasks. That'd allow us to simply hook into the compilation step like a normal plugin (instead of overriding the task entirely).

Reply:
Thanks for the feedback, @kfichter! We are currently going through an overhaul of the compilation pipeline, so we'll take your ideas into account. If you want to follow it, or propose any other thing, here's the issue track ihttps://github.com/nomiclabs/buidler/issues/553